File: migratoryBirds/migratoryBirds.js
const migratoryBirds = (arr) => {
  let birdCount = {};
  let commonBird = 0;
  let maxCount = 0;

  for (let bird of arr) {
    if (!birdCount[bird]) {
      birdCount[bird] = 1;
    } else {
      birdCount[bird] += 1;
    }
    if (birdCount[bird] > maxCount) {
      maxCount = birdCount[bird];
      commonBird = bird;
    } else if (birdCount[bird] == maxCount && commonBird > bird) {
      commonBird = bird;
    }
  }
  return commonBird;
};

const birds = [1, 4, 4, 5, 3, 3, 3];
console.log(migratoryBirds(birds));

File: Week 1 2 3/migratoryBirds/MigratoryBirds.java
package migratoryBirds;

import java.util.*;


public class MigratoryBirds{
    public static void main (String[] args){
    List<Integer> Birds = new ArrayList<>();
    Birds.addAll(Arrays.asList(1, 4, 4, 5, 3, 3, 3));
    System.out.println(migratoryBirds(Birds));
    }
    public static Number migratoryBirds(List<Integer> arr){
        Map<Integer, Long> birdCount = new HashMap<>();
    Integer commonBird = 0;
    Long maxCount = 0L;
    
    for (Integer bird : arr) {
        birdCount.put(bird, birdCount.getOrDefault(bird, 0L) + 1);

        if (birdCount.get(bird) > maxCount) {
            maxCount = birdCount.get(bird);
            commonBird = bird;
        } else if (birdCount.get(bird).equals(maxCount) && commonBird > bird) {
            commonBird = bird;
        }
    }
    return commonBird;
    }
}